geniza/corpus/models.py

```diff
@@ -527,6 +527,14 @@ def shelfmark_display(self):
             return None
         return certain[0] + (" + …" if len(certain) > 1 else "")
 
+    @property
+    def historical_date_display(self):
+        if self.doc_date_calendar:
+            calendar_name = dict(self.CALENDAR_CHOICES)[self.doc_date_calendar]
+            return f"{self.doc_date_original} {calendar_name}"
+        else:
+            return self.doc_date_original
+
```

Comment:
Django automatically provides a display method for choice fields, please use that instead.

I think you can simplify like this:
```suggestion
        return " ".join([self.doc_date_original, self.get_doc_date_calendar_display()]).strip()
```